hack/run-lint.sh

```diff
@@ -23,7 +23,7 @@ gopath="$(go env GOPATH)"
 if ! [[ -x "$gopath/bin/golangci-lint" ]]; then
   echo >&2 'Installing golangci-lint'
   curl --silent --fail --location \
-    https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b "$gopath/bin" v1.44.0
+    https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$gopath/bin" v1.44.0
```

Comment:
let's just do `go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.46.2` as the project page suggests?

Reply:
Note that the project **doesn't** recommend it:
> Install from Source
> Note: such go install/go get installation aren't guaranteed to work. We recommend using binary installation.

Anyway, pushed the change via [d80edeb](https://github.com/kubernetes-sigs/krew/pull/784/commits/d80edebbc5c4c1695661709adbdd1338e3ae535e) (it's a bit slower than the binary installation, comparing https://github.com/kubernetes-sigs/krew/runs/6596083051?check_suite_focus=true and https://github.com/kubernetes-sigs/krew/runs/6592749482?check_suite_focus=true).
Didn't update the version because some newer versions have a lint issue, see https://github.com/kubernetes-sigs/krew/issues/778#issuecomment-1125986802.

Reply:
we can revert back to `    https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$gopath/bin" v1.44.0` if you want.

Reply:
I'm new here and don't have a strong opinion, but using the precompiled binary seems enough on CI. Dropped the commit but if someone prefers `go install` one, I'm happy to push back it :)